### PR TITLE
[Reviewer: Matt] Change +sip.phone to the clearer +sip.with-twin, and prevent native-only...

### LINKED ISSUE #14: Gemini could fork twice to the same contact
When Gemini forks a call it includes `Accept-Contact: g.3gpp.ics` on one fork and `Reject-Contact: +sip.phone` on the other.  If a contact registered with `g.3gpp.ics` but not `+sip.phone`, both of Gemini's forks would end up matching their contact details and they'd end up being rung twice.  Instead, Gemini should specify `Reject-Contact: g.3gpp.ics` on the `Reject-Contact: +sip.phone` fork.

### PR DESCRIPTION
**Pull request by rkd-msw**
... devices from ringing twice

Fixes #14 and #15. UTed but not live-tested - once you've reviewed, I'll make the corresponding clearwater-live-test changes and run those.

### CODE REVIEW

**Comment by mirw** [comment]
Looks good.  As discussed, for completeness I think it would be good to mark `Reject-Contact: g.3gpp.ics` on the `Accept-Contact: +sip.with-twin` branch too.

File: src/mobiletwinned.cpp [hunk 1]
```diff
@@ -140,17 +140,30 @@ void MobileTwinnedAppServerTsx::on_initial_request(pjsip_msg* req)
...
+  // +sip.with-twin to the request headed for the VoIP client. (We
+  // require that Gemini-compatible clients set this if they detect
+  // that they are colocated with a native client).
   LOG_DEBUG("Creating forked request to VoIP client");
   pj_pool_t* voip_pool = get_pool(voip_req);
   pjsip_reject_contact_hdr* reject_hdr =
                                    pjsip_reject_contact_hdr_create(voip_pool);
   pjsip_param* phone = PJ_POOL_ALLOC_T(voip_pool, pjsip_param);
-  phone->name = STR_PHONE;
+  phone->name = STR_WITH_TWIN;
```
Comment (bossmc) [inline src/mobiletwinned.cpp]: `phone` is a terrible variable name after the rename.

File: src/mobiletwinned.cpp [hunk 2]
```diff
@@ -140,17 +140,30 @@ void MobileTwinnedAppServerTsx::on_initial_request(pjsip_msg* req)
...
+                                   pjsip_reject_contact_hdr_create(voip_pool);
+  pjsip_param* reject_native = PJ_POOL_ALLOC_T(voip_pool, pjsip_param);
+  reject_native->name = STR_3GPP_ICS;
+  reject_native->value.slen = 0;
+  pj_list_insert_after(&reject_hdr2->feature_set, reject_native);
+  pjsip_msg_add_hdr(voip_req, (pjsip_hdr*)reject_hdr2);
+
   // Set up the fork to the native device.
   // Append the twin prefix (if set) to the request URI, and add an
   // Accept-Contact header specifying g.3gpp.ics.
```
Comment (bossmc) [inline src/mobiletwinned.cpp]: This needs `explicit` and `required` set (otherwise this fork goes to other VoIP endpoints).

**Comment by rkd-msw** [comment]
All markups made.

**Comment by bossmc** [comment]
How much more work would it be to fix #18 and #19 while you're in the area?

**Comment by rkd-msw** [comment]
I think those two would require clearwater-live-test updates, so I'll do them separately. OK?

**Comment by bossmc** [comment]
Fine by me, just trying to be efficient :-)

**Pull request merged**